### PR TITLE
ARROW-5943: [GLib][Gandiva] Add support for function aliases

### PR DESCRIPTION
**Pull request by kou**
This is follow-up for ARROW-5892: https://github.com/apache/arrow/pull/4835

### CODE REVIEW

File: c_glib/gandiva-glib/native-function.h [hunk 1]
```diff
@@ -51,7 +51,8 @@ struct _GGandivaNativeFunctionClass
   GObjectClass parent_class;
 };
 
-GGandivaFunctionSignature *ggandiva_native_function_get_signature(GGandivaNativeFunction *native_function);
+GList *
+ggandiva_native_function_get_signatures(GGandivaNativeFunction *native_function);
```
Comment (shiro615) [inline c_glib/gandiva-glib/native-function.h]: Do we add `GARROW_AVAILABLE_IN_1_0` in another pull request after #4841 is merged?

Comment (kou) [reply c_glib/gandiva-glib/native-function.h]: Yes.

Comment (shiro615) [reply c_glib/gandiva-glib/native-function.h]: OK.

**Review by shiro615 — APPROVED** [review]
LGTM

**Pull request closed without merging**